sessionctx/stmtctx/stmtctx.go

```diff
@@ -584,6 +585,15 @@ func (sc *StatementContext) SetFlagsFromPBFlag(flags uint64) {
 	sc.DividedByZeroAsWarning = (flags & model.FlagDividedByZeroAsWarning) > 0
 }
 
+// GetLockWaitStartTime returns the statement pessimistic lock wait start time
+func (sc *StatementContext) GetLockWaitStartTime() time.Time {
+	if sc.lockWaitStartTime == nil {
```

Comment:
I think we better set it in `ResetContextOfStmt`

Reply:
There will be some time between `ResetContextOfStmt` and `tikv.LockKeys` , and this diff will be added into lock wait time